### PR TITLE
Add documentation to zebra-utils and checkpoint generation

### LINKED ISSUE #1482: Add a README for checkpoint list generation
**Is your feature request related to a problem? Please describe.**

We want to regenerate the checkpoint lists regularly, but there's nothing in the `checkpoint` directory that explains how to do that.

**Describe the solution you'd like**

Add a `README.md` to `zebra-consensus/src/checkpoint` that explains how to use the `zebra-checkpoints` binary from `zebra-utils`.

We should link to the existing `zebra-checkpoints` instructions at `book/src/dev/zebra-checkpoints.md`.

**Describe alternatives you've considered**

Do nothing. Only a few of us will know how to regenerate checkpoint lists.

**Additional context**

We're also writing a README for all the `zebra-utils` in #1483

### LINKED ISSUE #1483: Add a README for zebra-utils
**Is your feature request related to a problem? Please describe.**

We have a few utilities in `zebra-utils`, but it's hard to get an overview of what they all do.

**Describe the solution you'd like**

Add a `README.md` to `zebra-utils`, which explains what each utility does.

We should link to the existing `zebra-checkpoints` instructions at `book/src/dev/zebra-checkpoints.md`.

**Describe alternatives you've considered**

Do nothing. Users and developers will continue to be confused about what the `zebra-utils` do.

**Additional context**

We're also writing a README for checkpoint list generation in #1482

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

Add missing documentation.

## Solution

The README for the checkpoints inside the `zebra-consensus` crate explains how to add new checkpoints.
The README inside `zebra-utils` will list and briefly explain the purpose and how to use the utilities we currently have.

## Review

Anyone can review, open to all kind of changes. 

## Related Issues

- close https://github.com/ZcashFoundation/zebra/issues/1482 
- close https://github.com/ZcashFoundation/zebra/issues/1483

### CODE REVIEW

File: book/src/dev/zebra-checkpoints.md [hunk 1]
```diff
@@ -5,4 +5,4 @@
 Developers should run this tool every few months to add new checkpoints for the `checkpoint_sync = true` mode.
 (By default, Zebra syncs up to Sapling using checkpoints. These checkpoints don't need to be updated.)
 
-See the source code for more details.
+For ore information on how to run this program visit [Zebra checkpoints document](https://github.com/ZcashFoundation/zebra/tree/main/zebra-consensus/src/checkpoint/README.md)
```
Comment (dconnolly) [inline book/src/dev/zebra-checkpoints.md]:
```suggestion
For more information on how to run this program visit [Zebra checkpoints document](https://github.com/ZcashFoundation/zebra/tree/main/zebra-consensus/src/checkpoint/README.md)
```

File: zebra-consensus/src/checkpoint/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Zebra checkpoints
+
+To speed up the initial chain sync, zebra uses a list of checkpoint hashes for the `Mainnet` and for the `Testnet`:
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/README.md]: Checkpointing can be slower, depending on your network connection:
```suggestion
Zebra validates pre-Sapling blocks using a list of `Mainnet` and `Testnet` block hash checkpoints:
```

File: zebra-consensus/src/checkpoint/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+- [Mainnet checkpoints](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/main-checkpoints.txt)
+- [Testnet checkpoints](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/test-checkpoints.txt)
+
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/README.md:7]:
```suggestion
Zebra can also be configured to use these checkpoints after Sapling:
    [consensus]
    checkpoint_sync = true

```

File: zebra-consensus/src/checkpoint/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## Update checkpoints
+
+Checkpoint lists are distributed with zebra, maintainers should update them about every few months to get newer hashes. Here we explain how this process is done.
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/README.md]:
```suggestion
Checkpoint lists are distributed with Zebra, maintainers should update them about every few months to get newer hashes. Here we explain how this process is done.
```

File: zebra-consensus/src/checkpoint/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+### Use the `zebra-checkpoints` utility
+
+`zebra-checkpoints` is the program we use to collect checkpoints. Currently this program uses `zcash-cli` to get the hashes. `zcash-cli` must be available in your machine and it must be connected to a synchronized(mainnet or testnet) instance of `zcashd` to get the most recent hashes.
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/README.md]:
```suggestion
`zebra-checkpoints` is the program we use to collect checkpoints. Currently this program uses `zcash-cli` to get the hashes. `zcash-cli` must be available in your machine and it must be connected to a synchronized (Mainnet or Testnet) instance of `zcashd` to get the most recent hashes.
```

**Review by teor2345 — COMMENTED** [review]
Looks good, let's just add a bit of extra explanation

File: zebra-consensus/src/checkpoint/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+Zebra validates pre-Sapling blocks using a list of `Mainnet` and `Testnet` block hash checkpoints:
+
+- [Mainnet checkpoints](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/main-checkpoints.txt)
+- [Testnet checkpoints](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/test-checkpoints.txt)
+
+Zebra can also be configured to use these checkpoints after Sapling:
+```
+    [consensus]
+    checkpoint_sync = true
```
Comment (teor2345) [inline zebra-consensus/src/checkpoint/README.md]: Whitespace:
```suggestion
[consensus]
checkpoint_sync = true
```

**Review by teor2345 — DISMISSED** [review]
I'll just commit a whitespace tweak, then we can merge

**Review by teor2345 — APPROVED** [review]

**Pull request merged**